src/Compilers/Core/Portable/CommandLine/EditorConfig.cs

```diff
@@ -163,6 +163,71 @@ public static EditorConfig Parse(string text, string parentDirectory)
             return new EditorConfig(globalSection, namedSectionBuilder.ToImmutable(), parentDirectory);
         }
 
+        /// <summary>
+        /// Combine an editorconfig with an editorconfig nested in a subdirectory to form a new "effective"
+        /// editorconfig. "Nested" is defined as the parent directory being an ordinal prefix of the nested
+        /// directory.
+        /// </summary>
+        /// <remarks>
+        /// Editorconfig files are combined by applying all properties from the parent editorconfig, then
+        /// applying all properties from the nested editorconfig, with any conflicts resolving in favor
+        /// of the nested editorconfig. Any new sections in the nested editorconfig will appear at the
+        /// end of <see cref="EditorConfig.NamedSections" />.
+        /// </remarks>
+        public static EditorConfig Combine(EditorConfig nested, EditorConfig parent)
+        {
+            Debug.Assert(nested.Directory.StartsWith(parent.Directory, StringComparison.Ordinal));
```

Comment:
>StringComparison.Ordinal)); [](start = 71, length = 27)

Why are we treating paths ordinally here?

Reply:
Seems like the most portable way to treat them.